README.md

```diff
@@ -1,160 +1,529 @@
 
-## Manual
+1. Generate a secret
+2. Show a QR code for the user to scan in
+3. Authenticate the token
 
-### speakeasy.hotp(options) | speakeasy.counter(options)
+### Generating a key
 
-Calculate the one-time password using the counter-based algorithm, HOTP. Specify the key and counter, and receive the one-time password for that counter position. You can also specify a password length, as well as the encoding (ASCII, hexadecimal, or base32) for convenience. Returns the one-time password as a string.
+Use Speakeasy's key generator to get a key.
 
-Written to follow [RFC 4226](http://tools.ietf.org/html/rfc4226). Calculated with: `HOTP(K,C) = Truncate(HMAC-SHA-1(K,C))`
+```js
+var secret = speakeasy.generateSecret();
+// Returns an object with secret.ascii, secret.hex, and secret.base32.
+// Also returns secret.otpauth_url, which we'll use later.
+```
 
-#### Options
+This will generate a secret key of length 32, which will be the secret key for the user. We don't want to set this as the user's secret key just yet – we first want to verify their token for the first time. So, store one of the encodings for the secret, preferably `secret.base32`, somewhere temporary.
 
-* `key`: the secret key in ASCII, hexadecimal, or base32 format. `K` in the algorithm.
-* `counter`: the counter position (moving factor). `C` in the algorithm.
-* `length` (default `6`): the length of the resulting one-time password.
-* `encoding` (default `ascii`): the encoding of the `key`. Can be `'ascii'`, `'hex'`, or `'base32'`. The key will automatically be converted to ASCII.
+### Displaying a QR code
 
-#### Example
+Next, we'll want to display a QR code to the user so they can scan in the secret into their app. Google Authenticator and similar apps take in a QR code that holds a URL with the protocol `otpauth://`, which you get automatically from `secret.otpauth_url`.
 
-```javascript
-// normal use.
-speakeasy.hotp({key: 'secret', counter: 582});
-// => 246642
+Use a QR code module to generate a QR code that stores the data in `secret.otpauth_url`, and then display the QR code to the user. This is one simple way to do it, which generates a PNG data URL which you can put into an `<img>` tag on a webpage:
 
-// use a custom length.
-speakeasy.hotp({key: 'secret', counter: 582, length: 8});
-// => 67246642
+```js
+// Use the node-qrcode package
+// npm install --save node-qrcode
+var QRCode = require('qrcode');
 
-// use a custom encoding.
-speakeasy.hotp({key: 'AJFIEJGEHIFIU7148SF', counter: 147, encoding: 'base32'});
-// => 974955
+// Get the data URL of the authenticator URL
+QRCode.toDataURL(secret.otpauth_url, function(err, data_url) {
+    console.log(data_url);
+});
+
+// Display this data URL to the user in an <img> tag
+// Example:
+write('<img src="' + data_url + '">');
```

Comment:
Yeah, I was trying to illustrate a multi-step process as if these pieces of code would be in different places, but obviously it doesn't make sense as a single block of code. Updated it to put this line in the callback in 37d4937122cfcd4cb96f64db3e371dfe424a4691